### PR TITLE
Account email verification

### PR DESCRIPTION
**Pull request by KhadijaMahanga**
## Description

- Very email on registration

Fixes # (issue)

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
Screenshot 2022-01-17 at 15 07 06

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: twoopstracker/authentication/urls.py [hunk 1]
```diff
@@ -1,10 +1,18 @@
+from dj_rest_auth.registration.views import ConfirmEmailView, VerifyEmailView
 from django.conf.urls import include
 from django.urls import path
 
 from .views import GoogleLogin
 
 urlpatterns = [
+    path("verify-email/", VerifyEmailView.as_view(), name="rest_verify_email"),
```
Comment (kilemensi) [inline twoopstracker/authentication/urls.py:8]: Isn't this part of `registration/` below?

File: twoopstracker/settings.py [hunk 2]
```diff
@@ -220,9 +220,19 @@
...
 ACCOUNT_EMAIL_REQUIRED = True
-ACCOUNT_EMAIL_VERIFICATION = "none"
+ACCOUNT_EMAIL_VERIFICATION = env.str("ACCOUNT_EMAIL_VERIFICATION", "none")
 ACCOUNT_USER_MODEL_USERNAME_FIELD = None
 ACCOUNT_USERNAME_REQUIRED = False
+ACCOUNT_CONFIRM_EMAIL_ON_GET = env.bool("TWOOPSTRACKER_DEBUG", False)
+LOGIN_URL = env("LOGIN_URL")
+
+
+EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
```
Comment (kilemensi) [inline twoopstracker/settings.py:230]: This is the [default](https://docs.djangoproject.com/en/3.2/ref/settings/#email-backend), no?

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Lets disable email verification for social accounts ([`SOCIALACCOUNT_EMAIL_VERIFICATION`](https://django-allauth.readthedocs.io/en/latest/configuration.html)).

File: .env.template [hunk 3]
```diff
@@ -14,3 +14,11 @@ TWOOPSTRACKER_ALLOWED_HOSTS=
 TWOOPSTRACKER_DEBUG=False
 TWOOPSTRACKER_CORS_ALLOWED_ORIGINS=
 TWOOPSTRACKER_SENTRY_DSN=
+ACCOUNT_CONFIRM_EMAIL_ON_GET=False
```
Comment (esirK) [inline .env.template:17]: We can use the `TWOOPSTRACKER_` naming just for consistency.

Comment (KhadijaMahanga) [reply .env.template:17]: @esirK these are package related settings

File: twoopstracker/authentication/urls.py (hunk 1 above)
Comment (KhadijaMahanga) [reply twoopstracker/authentication/urls.py:8]: @kilemensi yes, you're right!

**Comment by kilemensi** [comment]
Why is the CI build failing @KhadijaMahanga ?

**Comment by KhadijaMahanga** [comment]
> Why is the CI build failing @KhadijaMahanga ?

@kilemensi I have been looking for solutions but nothing till now. It fails on `Run Lint`. But `make lint` passes on my local.

File: .env.template (hunk 3 above)
Comment (esirK) [reply .env.template:17]: I see. 
Although I meant something like 
```
TWOOPSTRACKER_ACCOUNT_CONFIRM_EMAIL_ON_GET=False
```
in the env file and then in the settings file we can have 

```
ACCOUNT_CONFIRM_EMAIL_ON_GET=env.bool("TWOOPSTRACKER_ACCOUNT_CONFIRM_EMAIL_ON_GET", False)
```
although I don't think this is a big deal.

**Comment by esirK** [comment]
> > Why is the CI build failing @KhadijaMahanga ?
> 
> @kilemensi I have been looking for solutions but nothing till now. It fails on `Run Lint`. But `make lint` passes on my local.

@KhadijaMahanga Last time this happened I had to debug it using something like [this](https://github.com/marketplace/actions/debugging-with-ssh). When you do the ssh you can try to run the lining from github actions and see why it's failing.

File: twoopstracker/authentication/urls.py [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
 from .views import GoogleLogin
 
 urlpatterns = [
+    path(
+        "account-confirm-email/",
+        VerifyEmailView.as_view(),
+        name="account_email_verification_sent",
+    ),
     path("google/", GoogleLogin.as_view(), name="google_login"),
+    path("registration/account-confirm-email/<str:key>/", ConfirmEmailView.as_view()),
```
Comment (kilemensi) [inline twoopstracker/authentication/urls.py:14]: Do we need this? As in clicking the link in verification email should take you to the _front-end_ and the front-end should be the one extracting the key and posting it to `/registration/verify-email/` API end-point.

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: twoopstracker/authentication/urls.py (hunk 4 above)
Comment (KhadijaMahanga) [reply twoopstracker/authentication/urls.py:14]: yes, that's sound right.

**Comment by KhadijaMahanga** [comment]
@kilemensi trolltracker.investigate.africa is the frontend URL. The email template is using `site.name` & `site.domain`.  I those in the admin

**Comment by kilemensi** [comment]
> @kilemensi trolltracker.investigate.africa is the frontend URL. The email template is using `site.name` & `site.domain`. I those in the admin

Ahhh... cool @KhadijaMahanga; I see investigate.africa I think API.

**Pull request merged**